### DISCUSSION TITLE
how feasible is to use EdgeDB as a replacement for SQLite?

### DISCUSSION THREAD

**Discussion by elzorrorebelde** [geldata/gel#2589] [Q&A] [answered]
Just wondering this, how feasible is to "deploy" a EdgeDB server locally to use as if it were SQLite? I noticed that this project is mostly Python and a little of Rust, and for Python you usually want the Python interpreter available when you intend to run a Python project.

The use case I'm visualizing is games and software that may require the convenience of a database and a query language to operate on it but may want to avoid SQL and NoSQL all together for something more convenient like EdgeDB.

**Comment by 1st1** [comment, marked as answer]
> Just wondering this, how feasible is to "deploy" a EdgeDB server locally to use as if it were SQLite?

The footprint of EdgeDB is going to be a lot bigger than of SQLite. The latter is purposefully built to be embeddable in applications. It should be possible to do with EdgeDB too, just going to be a bit more complicated.

> I noticed that this project is mostly Python and a little of Rust, and for Python you usually want the Python interpreter available when you intend to run a Python project.

All EdgeDB packages bundle their own Python interpreter, we never depend on the system Python. So at least this point shouldn't be a concern!

**Reply by elzorrorebelde** [reply]
in my mind, being able to run the EdgeDB server from a standalone executable is good enough, SQLite is made to be embedded indeed, but I don't intend that with EdgeDB.

Is there any documentation on how to accomplish this with EdgeDB?

**Reply by 1st1** [reply]
> Is there any documentation on how to accomplish this with EdgeDB?

Unfortunately no, not at this point, unfortunately. That said, if the application has docker as a dependency then packaging EdgeDB along with it would be trivial.

**Comment by elzorrorebelde** [comment]
well, I was more interested in games. Guess I'll have to build my abstraction over SQLite and see what happens

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by stereobooster** [comment]
I wonder if something changed since this issue was opened. As of now EdgeDB described as "Turbocharged Postgres
data layer". Can it be combined with https://pglite.dev/ to get embedded version of EdgeDB?
